test/e2e/suite/issuers/acme/certificate/http01.go

```diff
@@ -277,6 +279,77 @@ var _ = framework.CertManagerDescribe("ACME Certificate (HTTP01)", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("should allow updating the dns name of a failing certificate that had a wrong dns name", func() {
+		certClient := f.CertManagerClientSet.CertmanagerV1().Certificates(f.Namespace.Name)
+
+		By("Creating a failing Certificate")
+		cert := gen.Certificate(certificateName,
+			gen.SetCertificateSecretName(certificateSecretName),
+			gen.SetCertificateIssuer(cmmeta.ObjectReference{Name: issuerName}),
+			gen.SetCertificateDNSNames("google.com"),
+		)
+		cert.Namespace = f.Namespace.Name
+
+		_, err := certClient.Create(context.TODO(), cert, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Making sure the Order failed with a 400 since google.com is invalid")
+		order := &cmacme.Order{}
+		err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
+			orders, err := listOwnedOrders(f.CertManagerClientSet, cert)
+			Expect(err).NotTo(HaveOccurred())
+
+			if len(orders) != 1 {
```

Comment:
I just realized that len(orders) > 1 is not enough; the replacement would be:

```diff
-			if len(orders) != 1 {
+			if len(orders) == 0 || len(orders) > 1 {
```

otherwise I might be able to go to the next instruction and have if `len(orders) == 0`, which means panic 😅